src/main/java/it/gov/pagopa/cgn/portal/facade/ProfileFacade.java

```diff
@@ -41,16 +44,31 @@
     @Transactional(Transactional.TxType.REQUIRED)
     public ResponseEntity<Profile> updateProfile(String agreementId, UpdateProfile updateProfile) {
         ProfileEntity profileEntity = updateProfileConverter.toEntity(updateProfile);
+        ProfileEntity dbProfile = profileService.getProfileFromAgreementId(agreementId);
+        if (!profileEntity.getSalesChannel().equals(dbProfile.getSalesChannel())
+                || !profileEntity.getDiscountCodeType().equals(dbProfile.getDiscountCodeType())) {
+            // if sales channel is changed we should unpublish all the discount of this profile
+            discountService
```

Comment:
While performing this operation a suspend email notification is sent to merchant for each suspended discount. Is it the correct behaviour?

Reply:
I think that it is the correct behaviour because we said we want to suspend any discount and warn the operator about this also in the frontend when changing the discount type or sale channel. 
Sending an email for every suspended discount makes this more clear and strong in my opinion and the operator can have a reminder of every suspended discount to update and publish.

Reply:
cc @Loreau